Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -413,102 +419,94 @@ namespace AzToolsFramework
             return false;
         }
 
-        // if the field is not empty then we have a file linked to the prefab.
+        // record an undo step if a local view bookmark component is added and configured
+        ScopedUndoBatch undoBatch("SetupLocalViewBookmarks");
+        undoBatch.MarkEntityDirty(bookmarkComponent->GetEntityId());
+
+        // initialize default locations to 0. This is a temporary solution to match the 12 locations of the legacy system
+        // once there is a UI for the view bookmarks these lines should be removed
+        auto createDefaultBookmarksFn = [this, &registry]()
+        {
+            if (m_localBookmarks.empty())
+            {
+                for (int i = 0; i < DefaultViewBookmarkCount; i++)
+                {
+                    AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(i);
+                    
+                    if (const bool succeed = registry->SetObject(finalPath, ViewBookmark());
+                        !succeed)
```

Comment:
nit: if the function already returns a boolean seems like a waste to assign it to a variable to then check that variable.

````
if (!registry->SetObject(finalPath, ViewBookmark()))
````

Reply:
Good call, now changed!